src/_igraph/graphobject.c

```diff
@@ -2975,6 +2975,59 @@ PyObject *igraphmodule_Graph_Realize_Degree_Sequence(PyTypeObject *type,
 }
 
 
+/** \ingroup python_interface_graph
+ * \brief Generates a graph with a specified degree sequence
+ * \return a reference to the newly generated Python igraph object
+ * \sa igraph_realize_bipartite_degree_sequence
+ */
+PyObject *igraphmodule_Graph_Realize_Bipartite_Degree_Sequence(PyTypeObject *type,
+                                 PyObject *args, PyObject *kwds) {
+
+  igraph_vector_int_t degrees1, degrees2;
+  igraph_edge_type_sw_t allowed_edge_types = IGRAPH_SIMPLE_SW;
+  igraph_realize_degseq_t method = IGRAPH_REALIZE_DEGSEQ_SMALLEST;
+  PyObject *degrees1_o, *degrees2_o;
+  PyObject *edge_types_o = Py_None, *method_o = Py_None;
+  igraphmodule_GraphObject *self;
+  igraph_t g;
+
+  static char *kwlist[] = { "degrees1", "degrees2", "allowed_edge_types", "method", NULL };
+  if (!PyArg_ParseTupleAndKeywords(args, kwds, "O|OOO", kwlist,
+                                   &degrees1_o, &degrees2_o, &edge_types_o, &method_o))
+    return NULL;
+
+  /* allowed edge types */
+  if (igraphmodule_PyObject_to_edge_type_sw_t(edge_types_o, &allowed_edge_types))
+    return NULL;
+
+  /* methods */
+  if (igraphmodule_PyObject_to_realize_degseq_t(method_o, &method))
+    return NULL;
+
+  /* First degree vector */
+  if (igraphmodule_PyObject_to_vector_int_t(degrees1_o, &degrees1))
+    return NULL;
+
+  /* Second degree vector */
+  if (igraphmodule_PyObject_to_vector_int_t(degrees2_o, &degrees2))
+    return NULL;
```

Comment:
At this point you already have an initialized `igraph_vector_int_t` in `degrees1` so you must destroy it before returning `NULL`. (There's no FINALLY stack in the Python interface, you need to do stuff manually).

Reply:
Thanks, fixed!